base_rest/restapi.py

```diff
@@ -35,18 +35,25 @@ def method(
                     the `to_response` method with this result and then return
                     the result of this call.
       :param auth: The type of authentication method
-      :param cors: The Access-Control-Allow-Origin cors directive value.
+      :param cors: The Access-Control-Allow-Origin cors directive value. When
+                   set, this automatically adds OPTIONS to allowed http methods
+                   so the Odoo request handler will accept it.
       :param bool csrf: Whether CSRF protection should be enabled for the route.
                         Defaults to ``False``
 
     """
 
     def decorator(f):
         _routes = []
-        for paths, http_method in routes:
+        for paths, http_methods in routes:
             if not isinstance(paths, list):
                 paths = [paths]
-            _routes.append(([p for p in paths], http_method))
+            if not isinstance(http_methods, list):
+                http_methods = [http_methods]
+            if cors and "OPTIONS" not in http_methods:
+                http_methods.append("OPTIONS")
```

Comment:
hm, this should better be
```suggestion
                http_methods = http_methods + ["OPTIONS"]
```